command-line/lib/Command/Address/Reward.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Address.Reward
+    ( Cmd
+    , mod
+    , run
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address
+    ( NetworkTag (..), bech32 )
+import Cardano.Address.Style.Shelley
+    ( mkNetworkDiscriminant )
+import Fmt
+    ( build, fmt )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, header, helper, info, progDesc )
+import Options.Applicative.Discrimination
+    ( networkTagOpt )
+import Options.Applicative.Help.Pretty
+    ( bold, indent, string, vsep )
+import Options.Applicative.Style
+    ( Style (..) )
+import System.Exit
+    ( die )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( hGetXPub, progName )
+
+import qualified Cardano.Address.Style.Shelley as Shelley
+import qualified Data.ByteString.Char8 as B8
+import qualified Data.Text.Encoding as T
+
+
+newtype Cmd = Cmd
+    { networkTag :: NetworkTag
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "stake" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Create a stake address"
+        <> header "Create a stake address \
+            \that references a staking key (1-1)."
+        <> footerDoc (Just $ vsep
+            [ string "The public key is read from stdin."
+            , string ""
+            , string "Example:"
+            , indent 2 $ bold $ string $ "$ "<>progName<>" recovery-phrase generate --size 15 \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key from-recovery-phrase Shelley > root.prv"
+            , indent 2 $ string ""
+            , indent 2 $ bold $ string "$ cat root.prv \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key child 1852H/1815H/0H/2/0 > stake.prv"
+            , indent 2 $ string ""
+            , indent 2 $ bold $ string "$ cat stake.prv \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" key public \\"
+            , indent 4 $ bold $ string $ "| "<>progName<>" address stake --network-tag 0"
+            , indent 2 $ string "addr1uqly0fjvrgguywze067gwhsexggtj8rrdnxczgp5vexe8zgxqns3g"
+            ])
+  where
+    parser = Cmd
+        <$> networkTagOpt Shelley
+
+run :: Cmd -> IO ()
+run Cmd{networkTag} = do
+    xpub <- hGetXPub stdin
+    case (mkNetworkDiscriminant . fromIntegral . unNetworkTag) networkTag of
+        Left e -> die (fmt $ build e)
+        Right discriminant -> do
+            let addr = Shelley.stakeAddress discriminant (Shelley.liftXPub xpub)
+            B8.hPutStr stdout $ T.encodeUtf8 $ bech32 addr
```

Comment:
This should use [`bech32With`](https://input-output-hk.github.io/cardano-addresses/haddock/cardano-addresses-1.0.0/Cardano-Address.html#v:bech32With) and provide an appropriate prefix:

- `stake` for mainnet 
- `stake_test` for testnet

Reply:
And what about values outside of 0 and 1?

Reply:
I just use `addr` for these.

Reply:
> And what about values outside of 0 and 1?

Right now they should be `stake`. `0` is testnet, and `1` designates the _current_ mainnet. Values beyond 1 are reserved for future mainnets. There are none currently.